packages/react-components/react-toast/src/state/vanilla/createToaster.ts

```diff
@@ -0,0 +1,159 @@
+import { createPriorityQueue } from '@fluentui/react-utilities';
+import { Toast, ToasterOptions, ToastId, ToastOptions, UpdateToastEventDetail } from '../types';
+
+function assignDefined<T extends object>(a: Partial<T>, b: Partial<T>) {
+  // This cast is required, as Object.entries will return string as key which is not indexable
+  for (const [key, prop] of Object.entries(b) as [keyof T, T[keyof T]][]) {
+    // eslint-disable-next-line eqeqeq
+    if (prop != undefined) {
+      a[key] = prop;
+    }
+  }
+}
+
+/**
+ * Toast are managed outside of the react lifecycle because they can be
+ * dispatched imperatively. Therefore the state of toast visibility can't
+ * really be managed properly by a declarative lifecycle.
+ */
+export function createToaster(options: Partial<ToasterOptions>) {
+  const { limit = Number.POSITIVE_INFINITY } = options;
+  const visibleToasts = new Set<ToastId>();
+  const toasts = new Map<ToastId, Toast>();
+  const defaulToastOptions: Pick<
+    ToastOptions,
+    'priority' | 'pauseOnHover' | 'pauseOnWindowBlur' | 'position' | 'timeout' | 'politeness' | 'onStatusChange'
+  > = {
+    onStatusChange: undefined,
+    priority: 0,
+    pauseOnHover: false,
+    pauseOnWindowBlur: false,
+    position: 'bottom-end',
+    timeout: 3000,
+    ...options,
+  };
```

Comment:
nit: As `defaulToastOptions` is used once, there is sense to move out `defaulToastOptions` to a constant outside the factory and the merge during toast creation:

```ts
const toast: Toast = {
   ...DEFAULT_TOAST_OPTIONS,
   ...options,
   // ...etc.
}
```